tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/PublicAssetSetMethod.scala

```diff
@@ -0,0 +1,133 @@
+    } yield {
+      InvocationWithContext(
+        invocation = Invocation(
+          methodName = methodName,
+          arguments = Arguments(
+            PublicAssetSerializer.serializePublicAssetSetResponse(PublicAssetSetResponse(
+              accountId = request.accountId,
+              oldState = Some(UuidState.INSTANCE),
+              newState = UuidState.INSTANCE,
+              created = Some(creationResults.retrieveCreated).filter(_.nonEmpty),
+              notCreated = Some(creationResults.retrieveErrors).filter(_.nonEmpty))).as[JsObject]),
+          methodCallId = invocation.invocation.methodCallId),
+        processingContext = recordCreationIdInProcessingContext(creationResults, invocation.processingContext))
+    }
+
+  override def getRequest(mailboxSession: MailboxSession, invocation: Invocation): Either[Exception, PublicAssetSetRequest] =
+    PublicAssetSerializer.deserializePublicAssetSetRequest(invocation.arguments.value).asEitherRequest
+
+  private def recordCreationIdInProcessingContext(results: PublicAssetCreationResults,
+                                                  processingContext: ProcessingContext): ProcessingContext =
+    results.retrieveCreated.foldLeft(processingContext)({
+      case (processingContext, (creationId, result)) =>
+        processingContext.recordCreatedId(ClientId(creationId.id), ServerId(Id.validate(result.id.asString()).toOption.get))
+    })
+}
+
+object PublicAssetSetCreatePerformer {
+  val LOGGER: Logger = LoggerFactory.getLogger(classOf[PublicAssetSetCreatePerformer])
+}
+
+class PublicAssetSetCreatePerformer @Inject()(val publicAssetRepository: PublicAssetRepository,
+                                              val blobResolvers: BlobResolvers,
+                                              val publicAssetSetService: PublicAssetSetService) {
+
+  def createPublicAssets(mailboxSession: MailboxSession, request: PublicAssetSetRequest): SMono[PublicAssetCreationResults] =
+    SFlux.fromIterable(request.create.getOrElse(Map.empty))
+      .concatMap {
+        case (publicAssetCreationId: PublicAssetCreationId, publicAssetCreationRequestAsJson: JsObject) => parseCreate(publicAssetCreationRequestAsJson)
+          .fold(SMono.error(_), creationRequest => createPublicAssets(mailboxSession, creationRequest, request.accountId)
+            .map(publicAsset => PublicAssetCreationSuccess(publicAssetCreationId, PublicAssetCreationResponse.from(publicAsset))))
+          .onErrorResume(error => handleError(publicAssetCreationId, error))
+      }.collectSeq()
+      .map(PublicAssetCreationResults)
+
+  private def handleError(publicAssetCreationId: PublicAssetCreationId, error: Throwable): SMono[PublicAssetCreationResult] =
+    error match {
+      case e: PublicAssetException => SMono.just(PublicAssetCreationFailure(publicAssetCreationId, e))
+      case e: BlobNotFoundException => SMono.just(PublicAssetCreationFailure(publicAssetCreationId, e))
```

Comment:
Please log (info?) before recovering the error

Reply:
I already log it at: `com.linagora.tmail.james.jmap.publicAsset.PublicAssetCreationFailure#asPublicAssetSetError`
```
LOGGER.warn("Unexpected exception", exception)
```